queries.sql

```diff
@@ -134,3 +134,89 @@ COUNT(animals.owner_id)
 FROM animals
 INNER JOIN owners ON owners.id = animals.owner_id
 GROUP BY owners.full_name;
+
+/*Vet clinic database: add "join table" for visits*/
+-- Who was the last animal seen by William Tatcher?
+SELECT a.name, v.visit_date
+FROM animals a
+  JOIN visits v ON a.id = v.animal_id
+  JOIN vets ve ON v.vet_id = ve.id
+WHERE ve.name = 'William Tatcher'
+ORDER BY v.visit_date DESC
+LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+SELECT COUNT(*)
+FROM (
+        SELECT v.animal_id from visits v
+            JOIN vets ve ON v.vet_id = ve.id
+        WHERE ve.name = 'Stephanie Mendez'
+        GROUP BY v.animal_id
+    )as xx;
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT ve.name, s.name
+FROM vets ve
+  LEFT JOIN specializations sp ON ve.id = sp.vet_id
+  LEFT JOIN species s ON sp.species_id = s.id;
+
+  -- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+  SELECT a.name
+FROM animals a
+    JOIN visits v ON a.id = v.animal_id
+    JOIN vets as ve ON v.vet_id = ve.id
+WHERE ve.name = 'Stephanie Mendez'
+    AND v.visit_date BETWEEN 'April 1, 2020' AND 'August 30, 2020';
+
+-- What animal has the most visits to vets?
+SELECT COUNT(v.animal_id) as total_animal_visits,
+    a.name
+FROM visits v
+    JOIN animals a ON v.animal_id = a.id
+GROUP BY a.name
+ORDER BY total_animal_visits DESC
+LIMIT 1;
+
+-- Who was Maisy Smith's first visit?
+SELECT a.name
+FROM animals a
+    JOIN visits v ON a.id = v.animal_id
```

Comment:
- [ ] Similar as I said above 👆🏽 it is better to use:
```
SELECT animals.name, vistis.visit_date
  FROM visits
  JOIN animals ON animals.id = visits.animal_id
```
The rest of your code is OK, it is because the response will be different.

Reply:
Thanks for the suggestion.  I already included it into the queries.sql